src/coreclr/jit/lsraxarch.cpp

```diff
@@ -1885,21 +1890,30 @@ int LinearScan::BuildIntrinsic(GenTree* tree)
             break;
     }
     assert(tree->gtGetOp2IfPresent() == nullptr);
+
+// TODO-XARCH-AVX512 this is overly constraining register available as NI_System_Math_Abs
+// can be lowered to EVEX compatible instruction (the rest cannot)
+#if defined(TARGET_AMD64)
+    regMaskTP opRegMask = RBM_LOWSIMD;
```

Comment:
possibly have a macro like `#define DEFAULT_CANDIDATE_MASK` and set it to `RBM_LOWSIMD` vs. `RBM_NONE`?

Reply:
I'm not sure this will apply after the changes, since now we isolate to the `BuildEvexIncompatibleMask` function. Can make adjustments after I push if need be, but I think it'll be ok how it is.